src/groups/product_group.jl

```diff
@@ -20,13 +20,20 @@ one. This type is mostly useful for equipping the direct product of group manifo
     ProductGroup(manifold::ProductManifold)
 """
 function ProductGroup(manifold::ProductManifold)
-    if !all(M -> (is_decorator_group(M) === Val(true)), manifold.manifolds)
+    if !all(is_group_decorator, manifold.manifolds)
         error("All submanifolds of product manifold must be or decorate groups.")
     end
     op = ProductOperation()
     return GroupManifold(manifold, op)
 end
 
+function decorator_transparent_dispatch(::typeof(group_exp!), M::ProductGroup, q, X)
+    return Val(:transparent)
+end
+function decorator_transparent_dispatch(::typeof(group_log!), M::ProductGroup, X, q)
```

Comment:
I'm not quite sure why `group_log` is implemented on `ProductManifold` instead of `ProductGroup` but OK.

Reply:
Hm? These dispatches are for `ProductGroup` as it should be, and they should not be `ProductManifold` – what am I missing in this code? It should be implemented on ProductGroup.

I was actually thinking whether ProductGroup is indeed needed; if we do `ProductManifold(Group)` – shouldn't that suffice? Haven'T had time to think that through carefully though.

Reply:
> Hm? These dispatches are for `ProductGroup` as it should be, and they should not be `ProductManifold` – what am I missing in this code? It should be implemented on ProductGroup.

It's probably best not to touch it if it works :slightly_smiling_face: .


> I was actually thinking whether ProductGroup is indeed needed; if we do `ProductManifold(Group)` – shouldn't that suffice? Haven'T had time to think that through carefully though.

IIRC it's there to not confuse direct and semi-direct product groups.

Reply:
Yes, I did even not change much from the types. And thanks for the clarification, I did not have the semi in mind, then of course one needs this type.